src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -287,6 +290,33 @@ public IActionResult GetAdminToken()
             return Ok(requestHeaderToken);
         }
 
+        [HttpPost]
+        [Route("admin/warmup")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        [RequiresRunningHost]
+        public async Task<IActionResult> Warmup([FromServices] IScriptJobHost scriptHost, [FromServices] IScriptHostManager scriptHostManager, [FromServices] IHostIdProvider hostIdProvider)
+        {
+            // The function must be called warmup and should be of type warmup trigger
+            FunctionDescriptor function = scriptHost.GetFunctionOrNull(WarmUpConstants.FunctionName);
```

Comment:
nit: recommend moving this lookup logic into a ScriptHost.GetWarmupFunctionOrNull helper

Reply:
Do you think the ScriptHost should have the knowledge of `WarmupTrigger`? I thought from Script Host's point of view, it's nothing special and shouldn't have a special getter for it.

Reply:
I guess if I add it in the extension, it's probably fine and cleaner that way. I will do that. Thanks for the suggestion.